src/tests/topp/ProteomicsLFQ_7_out.mzTab

```diff
@@ -70,59 +70,59 @@ PRT	tr|A9FUD7|A9FUD7_SORC5	Putative membrane protein OS=Sorangium cellulosum (st
+PSM	YLYEIAR	9	P02769|ALBU_BOVIN	1	18Protein_SoCe_Tr_detergents_trace_target_decoy	null	[, , OMSSA, 2.1.4]	0.0	null	2321.499267578119998	2	464.250213623046989	464.250362519471025	ms_run[1]:spectrum=3328	K	R	null	null	17716858074023296841	0	spectrum=3328	YLYEIAR	internal	5.28980302830806e-04	0
+PSM	YLYEIAR	10	P02769|ALBU_BOVIN	1	18Protein_SoCe_Tr_detergents_trace_target_decoy	null	[, , OMSSA, 2.1.4]	0.0	null	2250.060546875	2	464.250061035156023	464.250362519471025	ms_run[2]:spectrum=3191	K	R	null	null	14891455924036538147	1	spectrum=3191	YLYEIAR	internal	1.18336398159532e-04	0
+PSM	KVPQVSTPTLVEVSR	11	P02769|ALBU_BOVIN	1	18Protein_SoCe_Tr_detergents_trace_target_decoy	null	[, , OMSSA, 2.1.4]	0.0	null	2281.44189453125	3	547.317077636718977	547.31743613990443	ms_run[2]:spectrum=3243	R	S	null	null	3246735940528705071	1	spectrum=3243	KVPQVSTPTLVEVSR	internal	1.60478309243352e-03	0
+PSM	LVVSTQTALA	12	P02769|ALBU_BOVIN	1	18Protein_SoCe_Tr_detergents_trace_target_decoy	null	[, , OMSSA, 2.1.4]	0.045454545454546	null	2431.521728515619998	2	501.794891357421989	501.795134726820947	ms_run[1]:spectrum=3482	K	null	null	null	2684106197423007927	0	spectrum=3482	LVVSTQTALA	internal	0.128894553541144	0
+PSM	LVVSTQTALA	13	P02769|ALBU_BOVIN	1	18Protein_SoCe_Tr_detergents_trace_target_decoy	null	[, , OMSSA, 2.1.4]	0.0	null	2341.02001953125	2	501.794799804688012	501.795134726820947	ms_run[2]:spectrum=3304	K	null	null	null	17569764383250687096	1	spectrum=3304	LVVSTQTALA	internal	0.018136079089243	0
+PSM	RPCFSALTPDETYVPK	14	P02769|ALBU_BOVIN	1	18Protein_SoCe_Tr_detergents_trace_target_decoy	null	[, , OMSSA, 2.1.4]	0.0	3-UNIMOD:4	2476.880615234380002	3	627.646362304687955	627.645224415137818	ms_run[2]:spectrum=3440	R	A	null	null	13113514917555180171	1	spectrum=3440	RPC(Carbamidomethyl)FSALTPDETYVPK	internal	5.78187927541962e-05	0
+PSM	KQTALVELLK	15	P02769|ALBU_BOVIN	1	18Protein_SoCe_Tr_detergents_trace_target_decoy	null	[, , OMSSA, 2.1.4]	0.0	null	2494.881591796880002	3	381.576354980468977	381.576298143604333	ms_run[2]:spectrum=3462	K	H	null	null	11275867043381640475	1	spectrum=3462	KQTALVELLK	internal	2.89327520051305e-10	0
+PSM	KQTALVELLK	16	P02769|ALBU_BOVIN	1	18Protein_SoCe_Tr_detergents_trace_target_decoy	null	[, , OMSSA, 2.1.4]	0.0	null	2492.93994140625	2	571.860717773437955	571.860808982021013	ms_run[2]:spectrum=3459	K	H	null	null	17447818880105247092	1	spectrum=3459	KQTALVELLK	internal	2.62096624298549e-05	0
+PSM	RHPEYAVSVLLR	17	P02769|ALBU_BOVIN	1	18Protein_SoCe_Tr_detergents_trace_target_decoy	null	[, , OMSSA, 2.1.4]	0.0	null	2412.920654296880002	3	480.608703613281023	480.608772593904405	ms_run[2]:spectrum=3369	R	L	null	null	1196373166564353753	1	spectrum=3369	RHPEYAVSVLLR	internal	9.077820128403891e-06	0
```

Comment:
- is there a way to make the order of the column with "internal" the same in both test files?
- is there a way to order the PSMs in the same way?

Reply:
at least we should have good reason to break potential down-stream scripts that hard-coded the column index for their data

Reply:
I don't understand why the column order would change. Meta values of PSMs are just copied from IdentificationData's ObservationMatches to PeptideHits, so there's no special treatment (https://github.com/hendrikweisser/OpenMS/blob/98656f05c5be92a6be822e65f6c6f450f37e4a83/src/openms/source/METADATA/ID/IdentificationDataConverter.cpp#L371).

The order of PSMs in IdentificationData is essentially random (based on memory addresses), so to ensure a consistent order, you'd need to sort them before output. This is done when exporting IdentificationData to mzTab directly (https://github.com/hendrikweisser/OpenMS/blob/98656f05c5be92a6be822e65f6c6f450f37e4a83/src/openms/source/METADATA/ID/IdentificationDataConverter.cpp#L724).

Down-stream scripts should check the order of columns in the header to avoid surprises.

Reply:
Do we need to change anything here before this can get merged?

Reply:
We need to be very sure that we don’t have (spike) memory usage or other regressions as we generate the packages for an ongoing large scale study using ProteomicsLFQ.